CMakeLists.txt

```diff
@@ -0,0 +1,180 @@
+endif ()
+
+#-----------------------------------------------------------------------
+# C++ tests.
+
+add_executable(test.cpp.core src/test.core.cpp)
+add_test(cpp.core test.cpp.core)
+target_link_libraries(test.cpp.core PRIVATE OpenAssetIO::openassetio-core)
+
+
+# GCC's C++11 ABI toggle must match OpenAssetIO's settings.
+if (DEFINED OPENASSETIOTEST_GLIBCXX_USE_CXX11_ABI)
+    if (OPENASSETIOTEST_GLIBCXX_USE_CXX11_ABI)
+        target_compile_definitions(test.cpp.core PRIVATE _GLIBCXX_USE_CXX11_ABI=1)
+    else ()
+        target_compile_definitions(test.cpp.core PRIVATE _GLIBCXX_USE_CXX11_ABI=0)
+    endif ()
+endif ()
+
+if (WIN32 AND DEFINED OpenAssetIO_BINARY_DIR)
+    set_tests_properties(cpp.core PROPERTIES ENVIRONMENT "PATH=${OpenAssetIO_BINARY_DIR_NATIVE}")
+endif ()
+
+#-----------------------------------------------------------------------
+# C tests.
+
+if (OPENASSETIOTEST_ENABLE_C)
+    add_executable(test.c.core src/test.core.c)
+    add_test(c.core test.c.core)
+    target_link_libraries(test.c.core PRIVATE OpenAssetIO::openassetio-core-c)
+    if (WIN32 AND DEFINED OpenAssetIO_BINARY_DIR)
+        set_tests_properties(c.core PROPERTIES ENVIRONMENT "PATH=${OpenAssetIO_BINARY_DIR_NATIVE}")
+    endif ()
+endif ()
+
+#-----------------------------------------------------------------------
+# Python tests.
+
+if (OPENASSETIOTEST_ENABLE_PYTHON)
+
+    if (OPENASSETIOTEST_ENABLE_SUBPROJECT)
+        message(
+            FATAL_ERROR
+            "Testing OpenAssetIO as a subproject when Python is enabled is currently unsupported."
+            " The binary Python module must be built and made available alongside the Python"
+            " sources for tests to work."
+        )
+    endif ()
+
+    find_package(Python COMPONENTS Interpreter Development.Embed REQUIRED)
+
+    if (NOT "${Python_VERSION}" STREQUAL "${OpenAssetIO_Python_VERSION}")
+        message(
+            FATAL_ERROR
+            "Found Python '${Python_VERSION}' but OpenAssetIO was built against"
+            " '${OpenAssetIO_Python_VERSION}'"
+        )
+    endif ()
+
+    #-----------------------------------------------------------------------
+    # Python sources tests.
+
+    # Python sources test.
+    add_test(NAME python COMMAND ${Python_EXECUTABLE} ${PROJECT_SOURCE_DIR}/src/test.py)
+    set_tests_properties(python PROPERTIES ENVIRONMENT PYTHONPATH=${OpenAssetIO_Python_SITELIB})
+
+    #-----------------------------------------------------------------------
+    # Python bridge test (loading Python plugin system from C++).
+
+    add_executable(test.python.bridge src/test.python.bridge.cpp)
+    add_test(python.cpp.bridge test.python.bridge)
+    target_link_libraries(
+        test.python.bridge
+        PRIVATE
+        OpenAssetIO::openassetio-python-bridge
+        Python::Python
+    )
+
+    # Storage for list of environment variables to set before running
+    # the test.
+    set(_envvars)
```

Comment:
Nit: do you know you don't need to start with an empty variable?

Reply:
I thought this would be best practice, in case at some point in the future `_envvars` was used above, so this would "reset" it back to blank.  I did toy with the idea of a really unique name like `python.cpp.bridge.envars` to guard against that instead, but thought it a bit wordy and obfuscating.

Reply:
> so this would "reset" it back to blank.

True

No harm in leaving, just wanted to make sure you knew that it wasn't strictly necessary.